python/ray/data/tests/test_actor_pool_map_operator.py

```diff
@@ -556,6 +594,113 @@ def __call__(self, x):
         ).take_all()
 
 
+def test_actor_pool_fault_tolerance_e2e(ray_start_cluster):
+    """Test that a dataset with actor pools can finish, when
+    all nodes in the cluster are removed and added back."""
+    ray.shutdown()
```

Comment:
this shutdown shouldn't be needed

Reply:
With this shutdown, ray.init() line 603 is throwing.

E               RuntimeError: Maybe you called ray.init twice by accident? This error can be suppressed by passing in 'ignore_reinit_error=True' or by calling 'ray.shutdown()' prior to 'ray.init()'.

Reply:
I think it's because the previously test didn't shutdown the cluster. We can change the `ray_start_regular_shared` to `ray_start_regular`